.github/workflows/release.yml

```diff
@@ -0,0 +1,251 @@
+      id: quay_tag
+      run: |
+        QUAY_TAG="${{ github.event.inputs.quay_repository }}:$RELEASE_VERSION"
+        
+        if [[ $RELEASE_TYPE == "minor" ]] || [ $RELEASE_TYPE == "patch" ]]
+        then
+          QUAY_TAG="$QUAY_TAG ${{ github.event.inputs.quay_repository }}:$BRANCH_VERSION"
+        fi
+
+        echo "::set-output name=quay_tag::$QUAY_TAG"
+    
+    - name: Log information
+      run: |
+        echo "Release type: ${{ steps.release_type.outputs.release_type }}"
+
+        echo "Release version: ${{ steps.release_version.outputs.release_version }}"
+                  
+        echo "Next version: ${{ steps.next_version.outputs.next_version }}"
+
+        echo "Branch version: ${{ steps.branch_version.outputs.branch_version }}"
+
+        echo "Quay tag": ${{ steps.quay_tag.outputs.quay_tag }}
+
+  build_backend:
+    name: Build backend
+    uses: ./.github/workflows/build-backend.yml
+    needs: [initialize]
+
+  build_frontend:
+    name: Build frontend
+    uses: ./.github/workflows/build-frontend.yml
+    needs: [initialize]
+    with:
+      target_branch: ${{needs.initialize.outputs.target_branch}}
+
+  # integration_tests_backend:
+  #   name: Run backend integration tests
+  #   uses: ./.github/workflows/integration-tests-backend.yml
+  #   needs: [initialize, build_backend, build_frontend]
+  #   with:
+  #     target_branch: ${{needs.initialize.outputs.target_branch}}
+
+  # integration_tests_frontend:
+  #   name: Run frontend integration tests
+  #   uses: ./.github/workflows/integration-tests-frontend.yml
+  #   needs: [initialize, build_backend, build_frontend]
+  #   with:
+  #     target_branch: ${{needs.initialize.outputs.target_branch}}
+
+  release:
+    name: Release
+    runs-on: ubuntu-20.04
+    #needs: [initialize, build_backend, build_frontend, integration_tests_backend, integration_tests_frontend]
+    needs: [initialize, build_backend, build_frontend]
+    env:
+      RELEASE_TYPE: ${{ needs.initialize.outputs.release_type }}
+      RELEASE_VERSION: ${{ needs.initialize.outputs.release_version }}
+      BRANCH_VERSION: ${{ needs.initialize.outputs.branch_version }}
+      NEXT_VERSION: ${{ needs.initialize.outputs.next_version }}
+      RELEASE_BRANCH: ${{ github.event.inputs.release_branch }}
+      QUAY_TAG: ${{ needs.initialize.outputs.quay_tag }}
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v3
+      with:
+        ref: ${{ github.event.inputs.release_branch }}
+
+    - name: Set version to release
+      run: sed -i -r "s/^VERSION \?= (.*)/VERSION \?= $RELEASE_VERSION/" Makefile
+
+    - name: Set up Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.17.7
+
+    - name: Cache Go deps
+      uses: actions/cache@v3
+      with:
+        path: |
+          ~/.cache/go-build
+          ~/go/pkg/mod
+        key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
+          ${{ runner.os }}-go-
```

Comment:
Late to comment here but why do we need `go` as part of the release pipeline? Since we already built the backend binary in a previous job, can we download that here instead of rebuilding the backend?

Reply:
AFAIU, the previous build is just for CI purposes (i.e. testing that it builds and fail fast). But that build is for amd64 arch. This workflow still needs to build the other arches.

Reply:
Ah right we need a separate build for each arch. The integration tests use that build to run the e2e tests against.